clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -4550,7 +4553,10 @@ static void handleLifetimeCategoryAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
     if (checkAttrMutualExclusion<PointerAttr>(S, D, AL))
       return;
     if (const auto *OAttr = D->getAttr<OwnerAttr>()) {
-      if (OAttr->getDerefType().getTypePtr() != ParmType.getTypePtr()) {
+      const Type *ExistingDerefType = OAttr->getDerefTypeLoc()
```

Comment:
Do we actually need to have the ternary operator? Could we just rely on comparing to QualTypes? Or two invalid QualTypes compared inequal? Then we could use getTypePtrOrNull on both of them.

Reply:
`getDerefType` dereferences a null pointer when `getDerefTypeLoc` is null.